pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -435,12 +435,26 @@ func preStopHookInit(volumeMount []corev1.VolumeMount) container.Modification {
 
 func mongodbContainer(version string, volumeMounts []corev1.VolumeMount) container.Modification {
 	mongoDbCommand := []string{
-		"/bin/sh",
+		"/bin/bash",
 		"-c",
 		// we execute the pre-stop hook once the mongod has been gracefully shut down by the agent.
 		`while [ ! -f /data/automation-mongod.conf ]; do sleep 3 ; done ; sleep 2 ;
+
+# terminate_mongod will send a SIGTERM signal to the mongod process
+terminate_mongod() {
+  kill -TERM "$mongo_pid" 2>/dev/null ;
```

Comment:
@fabianlindfors have you checked if the `kill` will wait for mongodb process to end? 
I don't remember the details and I could be wrong but while working on trap in mongodb enterprise I had to wait for the process to shutdown:
https://github.com/10gen/ops-manager-kubernetes/blob/6491bbfefc5e4f147198a7464bff76a207e389f2/docker/mongodb-enterprise-database/content/agent-launcher-lib.sh#L41

Reply:
You're right! I thought it was working but the process wasn't waiting but was running the pre-stop hook immediately. The pre-stop hook stayed alive long enough for the server to shut down gracefully but that's a horrible way of doing it.

I've updated the script to use a `wait` inside of of `terminate_mongod` which seems to work in our case. I also added an `exit` after that to stop the pre-stop hook from running. It just slows down the shut down procedure without doing anything.